File: C#/0437. Path Sum III.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int PathSum(TreeNode root, int sum) {
        int rep=0;
        if(root==null){
            return 0;
        }
        return NodeToPathSum(root,sum)+PathSum(root.left,sum)+PathSum(root.right,sum);
    }
    
    public int NodeToPathSum(TreeNode root, int sum){
        if(root==null){
            return 0;
        }
        else{
            int rep=0;
            if(sum==root.val){
                rep+=1;
            }
            return rep+NodeToPathSum(root.left,sum-root.val)+NodeToPathSum(root.right,sum-root.val);
            
        }
    }
}

File: archives/leetcode2/0437. Path Sum III.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def pathSum(self, root, sum):
        """
        :type root: TreeNode
        :type sum: int
        :rtype: int
        """
        if not root:
            return 0
        else:
            rep=self.pathSum(root.left,sum)+self.pathSum(root.right,sum)+self.allPathSum(root,sum)
            return rep
        
    def allPathSum(self,root,target):
        if not root:
            return 0
        else:
            rep=0
            if target==root.val:rep+=1
            rep+=self.allPathSum(root.left,target-root.val)+self.allPathSum(root.right,target-root.val)
            return rep
